lms/djangoapps/verify_student/views.py

```diff
@@ -1047,3 +1058,142 @@ def reverification_window_expired(_request):
+            return redirect(reverse('verify_student_verify_later', kwargs={'course_id': unicode(course_key)}))
+
+        # emit the reverification event
+        _track_reverification_events(
+            EVENT_NAME_USER_ENTERED_INCOURSE_REVERIFY_VIEW, user.id, course_id, checkpoint_name
+        )
+
+        context = {
+            'course_key': unicode(course_key),
+            "course_name": course.display_name_with_default,
+            'checkpoint_name': checkpoint_name,
+            'platform_name': settings.PLATFORM_NAME,
+        }
+        return render_to_response("verify_student/incourse_reverify.html", context)
+
+    @method_decorator(login_required)
+    def post(self, request, course_id, checkpoint_name):
+        """Submits the re-verification attempt to SoftwareSecure
+
+        Args:
+            request(HttpRequest): HttpRequest object
+            course_id(str): Course Id
+            checkpoint_name(str): Checkpoint name
+
+        Returns:
+            HttpResponse with status_code 400 if photo is missing or any error
+            or redirect to verify_student_verify_later url if initial verification doesn't exist otherwise
+            HttpsResponse with status code 200
+        """
+        # Check the in-course re-verification is enabled or not
+        incourse_reverify_enabled = InCourseReverificationConfiguration.current().enabled
+        if not incourse_reverify_enabled:
+            raise Http404
+        user = request.user
+        course_key = CourseKey.from_string(course_id)
+        course = modulestore().get_course(course_key)
+        checkpoint = VerificationCheckpoint.get_verification_checkpoint(course_key, checkpoint_name)
+        if checkpoint is None:
+            log.error("Checkpoint is not defined. Could not submit verification attempt for user %s",
+                      request.user.id)
+            context = {
+                'course_key': unicode(course_key),
+                "course_name": course.display_name_with_default,
+                'checkpoint_name': checkpoint_name,
+                "error": True,
+                "errorMsg": "No checkpoint found",
+                'platform_name': settings.PLATFORM_NAME,
+            }
+            return render_to_response("verify_student/incourse_reverify.html", context)
+        init_verification = SoftwareSecurePhotoVerification.get_initial_verification(user)
+        if not init_verification:
+            log.error("Could not submit verification attempt for user %s", request.user.id)
+            return redirect(reverse('verify_student_verify_later', kwargs={'course_id': unicode(course_key)}))
+
+        try:
+            b64_face_image = request.POST['face_image'].split(",")[1]
+            attempt = SoftwareSecurePhotoVerification(user=request.user)
+            attempt.upload_face_image(b64_face_image.decode('base64'))
+            attempt.photo_id_key = init_verification.photo_id_key
+            attempt.mark_ready()
```

Comment:
This seems like logic that we should be using elsewhere. Is there a utility function available for parsing and submitting images? If not, I'd suggest creating one and using it here.